otel-collector/pkg/quit/quit.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2023 Chainguard, Inc.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package quit
+
+import (
+	"log"
+	"net/http"
+	"time"
+)
+
+// Quit explicitly manages the lifecycle of the otel-collector sidecar. When a Job is
+// injected with an otel-collector sidecar, which is an always-on container, such Job
+// will never complete.
+//
+// This utility function sends a POST to otel-collector sidecar's /quitquitquit endpoint.
+// Jobs can use this to terminate the sidecar upon completion.
+func Quit() {
```

Comment:
Does this match the signature needed to be used with the new context OnClose (or w/e) function?

Reply:
Added a test